File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int m = s.length(), n = t.length();
        if (m != n) return false;

        unordered_map<char, int> umap;

        for(auto c : s) {
            umap[c]++;
        }

        for(auto c : t) {
            umap[c]--;
        }

        for(auto itr = umap.begin(); itr != umap.end(); ++itr) {
            if (umap[itr->first] != 0) {
                return false;
            }
        }

        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
            
        hashMap = {}

        for c in s:
            if c not in hashMap:
                hashMap[c] = 1
            else:
                hashMap[c] += 1
        
        for c in t:
            if c in hashMap:
                hashMap[c] -= 1
        
        print(hashMap)
        for key, value in hashMap.items():
            if value != 0:
                return False
        
        return True
        
